usr/src/uts/common/fs/zfs/zap_leaf.c

```diff
@@ -697,7 +688,7 @@ zap_entry_normalization_conflict(zap_entry_handle_t *zeh, zap_name_t *zn,
 			continue;
 
 		if (zn == NULL) {
-			zn = zap_name_alloc(zap, name, MT_FIRST);
+			zn = zap_name_alloc(zap, name, MT_FORM);
```

Comment:
I'm not clear on why this is MT_FORM vs MT_CASE vs based on some defaults in the ZAP.  It looks like this is used (with `zn==NULL`) only by `fzap_cursor_retrieve()`.